src/Backend/ContactManager.vala

```diff
@@ -67,6 +70,63 @@ public class Mail.ContactManager : GLib.Object {
         entry.set_completion (completion);
     }
 
+    public async void remember_mail_address (string address, string? name) {
+        name = name ?? address;
+
+        if (!Granite.Services.System.history_is_enabled ()) {
+            debug ("History disabled, don't remember email addresses.");
+            return;
+        }
+
+        if (address in email_addresses) {
```

Comment:
Is this totally necessary? add_persona_from_details already checks for existing details for us, and this list isn't utilized anywhere else.

Reply:
I know it says in the docs it does but with my testing it still creates additional entries that for example show up in GNOME Contacts... maybe I'm doing something wrong?

Reply:
I have been looking at the docs more and I wonder if what we should be doing is seeing if an individual exists with the details in question, and if they do use that as the parent in the `add_persona_from_details` call? That way if it is the same person but different email, the contact gets appended to instead of a new contact with the same name.